core/src/util/mod.rs

```diff
@@ -23,12 +23,24 @@ lazy_static! {
 }
 
 // Converts SQL pattern characters (% and _) into proper regex patterns.
+// SQL-1308: Handle SQL_ATTR_METADATA_ID
 // SQL-1060: Improve SQL-to-Rust regex pattern method
 // Returns regex for a filter
 pub(crate) fn to_name_regex(filter: &str) -> Option<Regex> {
     match filter {
-        "%" => None,
-        _ => Some(Regex::new(&filter.replace('%', ".*").replace('_', ".")).unwrap()),
+        "%" | "" => None,
+        _ => {
+            let filter = "^".to_owned() + &filter.replace('%', ".*").replace('_', ".") + "$";
```

Comment:
Does the ODBC spec allow for escaping? I don't think this will help our Power BI issue. What we really need to do is support the attribute that allows argument to metadata functions to be treated as literals instead of patterns (and probably set that by default for PowerBI).

Reply:
Some arguments in the catalog functions, such as the TableName argument in SQLTables, accept search patterns. These arguments accept search patterns if the SQL_ATTR_METADATA_ID statement attribute is set to SQL_FALSE; they are identifier arguments that do not accept a search pattern if this attribute is set to SQL_TRUE.

The search pattern characters are:

An underscore (_), which represents any single character.

A percent sign (%), which represents any sequence of zero or more characters.

An escape character, which is driver-specific and is used to include underscores, percent signs, and the escape character as literals. If the escape character precedes a non-special character, the escape character has no special meaning. If the escape character precedes a special character, it escapes the special character. For example, "\a" would be treated as two characters, "\" and "a", but "\%" would be treated as the non-special single character "%".

Reply:
ODBC supports escaping : https://learn.microsoft.com/en-us/sql/odbc/reference/develop-app/pattern-value-arguments?view=sql-server-ver16

The statement attribute SQL_ATTR_METADATA_ID is a global on-off switch, meaning that you can't use `my\_table_` to look for my_table1, my_table2, etc... for example.

I think it's a good idea though to see if PowerBI will generate escape sequences or not. My issue is that if not, then if we have 2 collections like `my_phone` and 'myiphone' then fields from both collections will still get mixed up as before, isn't it?

Reply:
I'll investigate this shortly. If it does, then I'll update the regex handling. If it doesn't, then we may need to do work on the connector.